File: proposed-projects/round-trip-experiment/reverse/20260909T123239Z_5982b634/20260910T070936Z_fb630842/ported_implementation/gbnf/grammar_graph/get_input_as_code_points.py
from __future__ import annotations

from typing import List

from .types import ValidInput


def get_code_point(char: str) -> int:
    if len(char) == 0:
        raise ValueError(f'Could not get code point for character: {char}')
    return ord(char[0])


def get_input_as_code_points(src: ValidInput) -> List[int]:
    if isinstance(src, bool):
        raise TypeError(f'Invalid input type: {type(src).__name__}')

    if isinstance(src, int):
        return [src]

    if isinstance(src, list):
        for code_point in src:
            if not isinstance(code_point, int) or isinstance(code_point, bool):
                raise TypeError(
                    f'code_point must be an integer for {code_point} if src is a list'
                )
        return src

    if isinstance(src, str):
        return [get_code_point(char) for char in src]

    raise TypeError(f'Invalid input type: {type(src).__name__}')

File: packages/gbnf-experiment/data/20260909T150124Z_a7cd2ee9/ported_implementation/src/grammar-graph/get-input-as-code-points.ts
import type { ValidInput } from './grammar-graph-types';

export const getCodePoint = (char: string): number => {
  const codePoint = char.codePointAt(0);
  if (codePoint === undefined) {
    throw new Error(`Could not get code point for character: ${char}`);
  }
  if (typeof codePoint !== 'number') {
    throw new Error('code_point must be an integer!');
  }
  return codePoint;
};

export const getInputAsCodePoints = (src: ValidInput): number[] => {
  if (typeof src === 'number') {
    return [src];
  }

  if (Array.isArray(src)) {
    for (const c of src) {
      if (typeof c !== 'number') {
        throw new Error(
          `code_point must be an integer for ${c} if src is a list`
        );
      }
    }
    return src;
  }

  if (typeof src === 'string') {
    return src.split('').map(getCodePoint);
  }

  throw new Error(`Invalid input type: ${typeof src}`);
};
